### DISCUSSION TITLE
Programming business logic with maui class library, and constructing UI with native frameworks.

### DISCUSSION THREAD

**Discussion by LITTOMA** [dotnet/maui#14484] [Ideas]
I am exploring options to program my business logic with a single code base, but I prefer not to use MAUI for UI development due to its limitations. 

In my experience, UIs created with native frameworks like SwiftUI and Jetpack Compose tend to be of higher quality. 

Has anyone else attempted this approach before?

**Comment by saint4eva** [comment]
> I am exploring options to program my business logic with a single code base, but I prefer not to use MAUI for UI development due to its limitations.
> 
> In my experience, UIs created with native frameworks like SwiftUI and Jetpack Compose tend to be of higher quality.
> 
> Has anyone else attempted this approach before?

The best place to ask this question is at SwiftUI or Jetpack Compose GitHub repo

**Reply by LITTOMA** [reply]
Great
